tenderly/unchainedProxy.ts

```diff
@@ -0,0 +1,145 @@
+/* eslint-disable @shapeshiftoss/logger/no-native-console */
+import type { evm } from '@shapeshiftoss/common-api'
+import dotenv from 'dotenv'
+import express from 'express'
+import { readFileSync } from 'fs'
+import { createProxyMiddleware, fixRequestBody } from 'http-proxy-middleware'
+import Web3 from 'web3'
+import type { Transaction } from 'web3-types'
+import { hexToNumber, numberToHex } from 'web3-utils'
+import type { RawData, WebSocket as WebSocketClient } from 'ws'
+import WebSocket, { WebSocketServer } from 'ws'
+
+import { corsMiddleware } from './middleware/cors'
+import {
+  PROXY_UNCHAINED_ETHEREUM_HTTP_PORT,
+  PROXY_UNCHAINED_ETHEREUM_WS_PORT,
+} from './utils/constants'
+import { getTenderlyRpcUrl } from './utils/getTenderlyRpcUrl'
+import { estimateGas, getBalance, getTransactionByHash, sendTransaction } from './utils/rpcHelpers'
+
+const { REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL, REACT_APP_UNCHAINED_ETHEREUM_WS_URL } = dotenv.parse(
+  readFileSync('.env.dev'),
+)
+
+const httpPort = Number(PROXY_UNCHAINED_ETHEREUM_HTTP_PORT)
+const websocketPort = Number(PROXY_UNCHAINED_ETHEREUM_WS_PORT)
+const provider = new Web3.providers.HttpProvider(getTenderlyRpcUrl())
+let subscriptionDetails: { subscriptionId: string; address: string } | undefined
+let clientWebsocket: WebSocketClient | undefined
+
+const captureSubscriptionDetails = (rawData: RawData) => {
+  const { subscriptionId, method, data } = JSON.parse(rawData.toString())
+
+  if (method !== 'subscribe' || data.topic !== 'txs') return
+
+  subscriptionDetails = { subscriptionId, address: data.addresses[0] }
+}
+
+const sendTxViaWebSocket = async (txid: string): Promise<void> => {
+  const result = await getTransactionByHash(provider, txid)
+
+  // for safety we coalesce to '0x0' for optionals but in practice should never be the case
+  const data: evm.Tx = {
+    txid,
+    blockHash: result.blockHash?.toString() ?? '0x0',
+    blockHeight: hexToNumber(result.blockNumber?.toString() ?? '0x0'),
+    timestamp: Date.now() / 1000, // assume now
+    from: result.from,
+    to: result.to!, // 'to' address is only missing for contract creation
+    confirmations: 10, // assume 10 confirmations
+    value: result.value?.toString() ?? '0x0',
+    fee: result.maxFeePerGas?.toString() ?? '0x0',
+    gasLimit: result.gas?.toString() ?? '0x0',
+    gasPrice: result.gasPrice?.toString() ?? '0x0',
```

Comment:
Curious about the `toString()` for `value` here, it is of `Numbers` type typed as `export declare type Numbers = number | bigint | string | HexString;` - is it coming in as an hex string? Is `value` missing for 0-value which is why we need to fallback to the 0 hex string?

Reply:
its actually never missing in any examples I've seen, but i've just implemented the same overcautious fallback logic for all optionals apart from `to` which is documented as only being missing for contract creation